velox/dwio/parquet/reader/ParquetReader.cpp

```diff
@@ -345,11 +331,23 @@ std::unique_ptr<ParquetTypeWithId> ReaderBase::getParquetColumnInfo(
       }
 
       TypePtr childRequestedType = nullptr;
+      bool followChild = true;
       if (requestedType && requestedType->isRow()) {
-        auto fileTypeIdx =
-            requestedType->asRow().getChildIdxIfExists(childName);
-        if (fileTypeIdx.has_value()) {
-          childRequestedType = requestedType->asRow().childAt(*fileTypeIdx);
+        auto rowType =
+            std::dynamic_pointer_cast<const velox::RowType>(requestedType);
+        if (options_.useColumnNamesForColumnMapping()) {
+          auto fileTypeIdx = rowType->getChildIdxIfExists(childName);
+          if (fileTypeIdx.has_value()) {
+            childRequestedType = rowType->childAt(*fileTypeIdx);
+          }
+        } else {
+          // Schema Evolution check
+          if (i < rowType->size()) {
```

Comment:
Just thinking. It is an error if we are missing a column in the middle of the schema. This will handle columns towards the end of the column list. What if the dropped column is the first one? Is that allowed? And if so would we end up in a mapping mismatch - thinking of missing on last columns because we are not allowed to look there?

Reply:
This path is taken if we are matching the schema by index.
In this case, the column types between the file and the table are matched in order. 
The schema evolution cannot happen in the middle. There is no way to identify incorrect schema otherwise.
If you want to evolve the schema in the middle, then you have to match the schema by name.